tiatoolbox/models/architecture/sccnn.py

```diff
@@ -0,0 +1,372 @@
+        Centroids of ground truth masks were used to train this model.
+        The results are reported on the whole test data set including preliminary
+        and final set.
+
+    The original model was implemented in Matlab. The model has been reimplemented
+    in PyTorch for Python compatibility. The original model uses HRGB as input,
+    where 'H' represents hematoxylin. The model has been modified to rely on RGB
+    image as input.
+
+
+    The tiatoolbox model should produce the following results on the following datasets
+    using
+    8 pixels as radius for true detection:
+
+    .. list-table:: MicroNet performance
+       :widths: 15 15 15 15 15
+       :header-rows: 1
+
+       * - Model name
+         - Data set
+         - Precision
+         - Recall
+         - F1Score
+       * - sccnn-crchisto
+         - CRCHisto
+         - 0.82
+         - 0.80
+         - 0.81
+       * - sccnn-crchisto
+         - CoNiC
+         - 0.79
+         - 0.79
+         - 0.79
+
+    Args:
+        num_input_channels (int):
+            Number of channels in input. default=3.
+        out_height (int):
+            Output height. default=13.
+        out_width (int):
+            Output width. default=13.
+        radius (int):
+            Radius for nucleus detection, default = 12.
+        min_distance (int):
+            The minimal allowed distance separating peaks.
+            To find the maximum number of peaks, use `min_distance=1`, default=6.
+        threshold_abs (float):
+            Minimum intensity of peaks, default=0.20.
+
+    References:
+        [1] Sirinukunwattana, Korsuk, et al.
+        "Locality sensitive deep learning for detection and classification
+        of nuclei in routine colon cancer histology images."
+        IEEE transactions on medical imaging 35.5 (2016): 1196-1206.
+
+    """
+
+    def __init__(
+        self,
+        num_input_channels: int = 3,
+        out_height: int = 13,
+        out_width: int = 13,
+        radius: int = 12,
+        min_distance: int = 6,
+        threshold_abs: float = 0.20,
+    ):
+        super().__init__()
+        self.in_ch = num_input_channels
+        self.out_height = out_height
+        self.out_width = out_width
+
+        # Create mesh grid and convert to 3D vector
+        x, y = torch.meshgrid(
+            torch.arange(start=0, end=out_height),
+            torch.arange(start=0, end=out_width),
+            indexing="ij",
+        )
+
+        self.register_buffer("xv", torch.unsqueeze(x, dim=0).type(torch.float32))
+        self.register_buffer("yv", torch.unsqueeze(y, dim=0).type(torch.float32))
+
+        self.radius = radius
+        self.min_distance = min_distance
+        self.threshold_abs = threshold_abs
+
+        def conv_act_block(
+            in_channels: int, out_channels: int, kernel_size: int
+        ) -> torch.nn.ModuleDict:
+            """Convolution and Activation branch for SCCNN.
+
+            Args:
```

Comment:
Is it possible to document a little bit what this block is / how it works / why it is used etc?

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/b30c6ebf15cf23c18c850aa2ac13b35180c30786/tiatoolbox/models/architecture/sccnn.py#L123